packages/solana-client/test/get-account-info.test.ts

```diff
@@ -0,0 +1,135 @@
+import { address } from '@solana/kit'
+import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest'
+import { getAccountInfo } from '../src/get-account-info.ts'
+import type { SolanaClient } from '../src/solana-client.ts'
+
+describe('get-account-info', () => {
+  describe('expected behavior', () => {
+    it('should call rpc.getAccountInfo with the correct address', async () => {
+      // ARRANGE
+      expect.assertions(2)
+      const testAddress = 'So11111111111111111111111111111111111111112'
+      const AccountInfo = {
+        value: {
+          data: new Uint8Array([1, 2, 3]),
+          executable: false,
+          lamports: 1000000n,
+          owner: address('11111111111111111111111111111111'),
+          rentEpoch: 0n,
+        },
+      }
+      const mockSend = vi.fn().mockResolvedValue(AccountInfo)
+      const mockGetAccountInfo = vi.fn().mockReturnValue({ send: mockSend })
+      const mockClient = {
+        rpc: {
+          getAccountInfo: mockGetAccountInfo,
+        },
+      } as unknown as SolanaClient
+
+      // ACT
+      const result = await getAccountInfo(mockClient, { address: testAddress })
+
+      // ASSERT
+      expect(mockGetAccountInfo).toHaveBeenCalledWith(address(testAddress))
+      expect(result).toEqual(AccountInfo)
+    })
+
+    it('should handle account with no data', async () => {
+      // ARRANGE
+      expect.assertions(1)
+      const testAddress = 'So11111111111111111111111111111111111111112'
+      const AccountInfo = {
+        value: null,
+      }
+      const mockSend = vi.fn().mockResolvedValue(AccountInfo)
+      const mockGetAccountInfo = vi.fn().mockReturnValue({ send: mockSend })
+      const mockClient = {
+        rpc: {
+          getAccountInfo: mockGetAccountInfo,
+        },
+      } as unknown as SolanaClient
```

Comment:
We should mock only what's needed but create the actual client https://vitest.dev/guide/mocking.html